src/devices/bus/epson_qx/option.h

```diff
@@ -0,0 +1,276 @@
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |---|  |---|  |---|  |---|  |---|
+
+    The DREQ/DACK(F) signals are the high priorty DMA signals and are shared between all option slots.
+    The DREQ/DACK(S) signals are the lower priority DMA signals and are local to each slot, with the
+    exception of slot 5 which does not provide low priority DMA signals. The RDY/EOP(S) signals are
+    also connected to all 5 options slots since the DREQ(S) signals are all connected to the same
+    DMA controller.
+
+      5      4      3      2      1
+    |---|  |---|  |---|  |---|  |---|
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------DREQF (Master CH3)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------DACKF (Master Ch3)
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------RDYF  (Master RDY)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------EOPF  (Master EOP)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  | *-|--------DREQS (Slave  CH0)
+    |   |  |   |  |   |  |   |  | *-|--------DACKS (Slave  CH0)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  | *-|--|---|--------DREQS (Slave  CH1)
+    |   |  |   |  |   |  | *-|--|---|--------DACKS (Slave  CH1)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  | *-|--|---|--|---|--------DREQS (Slave  CH2)
+    |   |  |   |  | *-|--|---|--|---|--------DACKS (Slave  CH2)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  | *-|--|---|--|---|--|---|--------DREQS (Slave  CH3)
+    |   |  | *-|--|---|--|---|--|---|--------DACKS (Slave  CH3)
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------RDYS  (Slave RDY)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------EOPS  (Slave EOP)
+    |   |  |   |  |   |  |   |  |   |
+    |---|  |---|  |---|  |---|  |---|
+
+*/
+#ifndef MAME_BUS_EPSON_QX_BUS_H
+#define MAME_BUS_EPSON_QX_BUS_H
+
+#pragma once
+
+#include <vector>
+
+//**************************************************************************
+//  FORWARD DECLARATIONS
+//**************************************************************************
+
+class option_bus_device;
+class device_option_expansion_interface;
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+/* Epson Slot Device */
+
+class option_slot_device : public device_t, public device_single_card_slot_interface<device_option_expansion_interface>
+{
+public:
+	friend class option_bus_device;
+	// construction/destruction
+	template <typename T, typename U>
+	option_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, T &&bus_tag, int slot, U &&opts, const char *dflt)
+		: option_slot_device(mconfig, tag, owner, 0)
```

Comment:
Given the clock is sent to the slots, I’d prefer that it wasn’t defaulted like this, so that cards can use it if anything on the card uses it for timings (cards inherit the slot’s clock by default).